packages/purpose-process/src/services/purposeService.ts

```diff
@@ -1020,19 +1023,26 @@ export function purposeServiceBuilder(
         purposeSeed.freeOfChargeReason
       );
 
-      const consumerDelegation =
-        await readModelService.getActiveConsumerDelegationByEserviceAndConsumerIds(
-          {
-            eserviceId,
-            consumerId,
-          }
-        );
+      // eslint-disable-next-line functional/no-let
+      let consumerDelegation: Delegation | undefined;
+
+      try {
+        assertRequesterIsConsumer({ consumerId }, authData);
+      } catch {
+        consumerDelegation =
+          await readModelService.getActiveConsumerDelegationByEserviceAndConsumerIds(
+            {
+              eserviceId,
+              consumerId,
+            }
+          );
 
-      assertRequesterCanActAsConsumer(
-        { eserviceId, consumerId },
-        authData,
-        consumerDelegation
-      );
+        assertRequesterIsDelegateConsumer(
+          { eserviceId, consumerId },
+          authData,
+          consumerDelegation
+        );
+      }
```

Comment:
The same shall be done for the reverse purpose creation (maybe as part of the dedicated test PR #1342)